### PR TITLE
STSMACOM-486 Guard appearance of remote-storage field with `hasInterface()

### PR DESCRIPTION
**Pull request by uladislausamets**
## Purpose

https://issues.folio.org/browse/STSMACOM-486
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by aliaksei-chumakou — APPROVED** [review]

File: lib/utils/useRemoteStorageMappings.js [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
+  const stripes = useStripes();
   const ky = useOkapiKy();
 
+  const withRemoteStorage = (
+    stripes.hasInterface('remote-storage-configurations') && stripes.hasInterface('remote-storage-mappings')
+  );
+
   const searchParams = { limit: LIMIT };
   const queryKey = [API, searchParams];
   const queryFn = () => ky(API, { searchParams }).json();
```
Comment (AndreiShumski2) [inline lib/utils/useRemoteStorageMappings.js:18]: `const queryFn = () => ky(API, { searchParams }, enabled: withRemoteStorage ).json();`
 So if !withRemoteStorage you don't need to make request.

File: lib/utils/useRemoteStorageMappings.js [hunk 2]
```diff
@@ -16,7 +21,10 @@ const useRemoteStorageMappings = () => {
 
   const records = data?.mappings ?? [];
 
-  return Object.fromEntries(records.map(r => [r.folioLocationId, r.configurationId]));
+  if (withRemoteStorage) {
```
Comment (axelhunn) [inline lib/utils/useRemoteStorageMappings.js:24]: Guard the query, and this is not gonna be needed

File: lib/utils/useRemoteStorageMappings.js [hunk 3]
```diff
@@ -1,16 +1,21 @@
...
+  const withRemoteStorage = (
+    stripes.hasInterface('remote-storage-configurations') && stripes.hasInterface('remote-storage-mappings')
+  );
+
   const searchParams = { limit: LIMIT };
   const queryKey = [API, searchParams];
-  const queryFn = () => ky(API, { searchParams }).json();
+  const queryFn = () => ky(API, { searchParams }, { enabled: withRemoteStorage }).json();
 
   const { data } = useQuery({ queryKey, queryFn });
```
Comment (AndreiShumski2) [inline lib/utils/useRemoteStorageMappings.js:20]: const { data } = useQuery({ queryKey, queryFn, enabled: withRemoteStorage });

**Review by AndreiShumski2 — APPROVED** [review]

File: lib/utils/useRemoteStorageMappings.js [hunk 4]
```diff
@@ -1,18 +1,23 @@
...
 const LIMIT = 1000;
 
 const useRemoteStorageMappings = () => {
+  const stripes = useStripes();
   const ky = useOkapiKy();
 
+  const withRemoteStorage = (
+    stripes.hasInterface('remote-storage-configurations') && stripes.hasInterface('remote-storage-mappings')
+  );
+
```
Comment (axelhunn) [inline lib/utils/useRemoteStorageMappings.js:15]:
```suggestion
  const enabled = stripes.hasInterface('remote-storage-configurations') && stripes.hasInterface('remote-storage-mappings');
```

File: lib/utils/useRemoteStorageMappings.js [hunk 5]
```diff
@@ -1,18 +1,23 @@
...
   const searchParams = { limit: LIMIT };
   const queryKey = [API, searchParams];
   const queryFn = () => ky(API, { searchParams }).json();
 
-  const { data } = useQuery({ queryKey, queryFn });
+  const { data } = useQuery({ queryKey, queryFn, enabled: withRemoteStorage });
```
Comment (axelhunn) [inline lib/utils/useRemoteStorageMappings.js:20]:
```suggestion
  const { data } = useQuery({ queryKey, queryFn, enabled });
```

**Review by axelhunn — APPROVED** [review]

File: package.json [hunk 6]
```diff
@@ -18,6 +18,10 @@
       "notes": "2.0",
       "tags": "1.0",
       "password-validator": "1.0"
+    },
+    "optionalOkapiInterfaces": {
+      "remote-storage-configurations": "1.0",
+      "remote-storage-mappings": "1.0"
```
Comment (axelhunn) [inline package.json]:
```suggestion
    },
    "optionalOkapiInterfaces": {
      "remote-storage-mappings": "1.0"
```

**Review by axelhunn — APPROVED** [review]

**Pull request merged**